src/main/java/com/google/cloud/tools/managedcloudsdk/update/SdkUpdater.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.managedcloudsdk.update;
+
+import com.google.cloud.tools.managedcloudsdk.MessageListener;
+import com.google.cloud.tools.managedcloudsdk.executors.SdkExecutorServiceFactory;
+import com.google.cloud.tools.managedcloudsdk.executors.SingleThreadExecutorServiceFactory;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.common.util.concurrent.ListeningExecutorService;
+import java.nio.file.Path;
+import java.util.concurrent.Callable;
+
+/** Update an SDK. */
+public class SdkUpdater {
```

Comment:
Since this class and the SdkComponentInstaller share a lot of common code it probably makes send to put all the common code in an abstract parent.